security/fma-interop-access-list.md

```diff
@@ -0,0 +1,121 @@
+# Interop Access List: Failure Modes and Recovery Path Analysis
+
+| Author              | Agusduha, Skeletor-spaceman  |
+| ------------------- | ---------------------------- |
+| Created at          | 2025-02-05                   |
+| Initial Reviewers   | [Initial Reviewer Names]     |
+| Needs Approval From | Matt Solomon, Kelvin Fichter |
+| Status              | Draft                        |
+
+## Introduction
+
+This document covers the failure modes and recovery paths for the access list validation mechanism in cross-chain message processing. The access list enhancement aims to prevent denial of service (DoS) attacks and enable cheap pre-validation of cross-chain messages.
+
+The following components are included:
+
+- **Contracts**:
+  - Updates to `CrossL2Inbox`: Implements gas introspection to validate `ExecutingMessage` transactions
+  - Storage slot warming mechanism for message validation
+  - Checksum calculation and verification logic
+
+Below are references for this project:
+
+- Design PR: [Access List design](https://github.com/ethereum-optimism/design-docs/pull/214)
+- Specs PR: [Access List Validation](https://github.com/ethereum-optimism/specs/pull/612)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Gas Schedule Changes Break Validation
+
+- **Description:** The validation mechanism relies on gas introspection to determine if storage slots are warm. If the EVM's gas schedule changes, particularly around storage access costs, it could break the validation logic.
+- **Risk Assessment:** High
+  - Potential Impact: Critical. Changes to gas costs could cause all cross-chain message validation to fail or allow invalid messages to pass.
+  - Likelihood: Low. While gas schedule changes are rare, they do occur during network upgrades.
+- **Mitigations:**
+  - Set `WARM_READ_THRESHOLD` conservatively to account for potential fluctuations
+  - Add tests that verify validation works across different gas cost scenarios
+  - Document gas schedule dependencies clearly for future upgrades
+- **Detection:** Monitor for network upgrades that modify storage access gas costs. Run validation tests before accepting upgrades.
+- **Recovery Path(s):** Deploy contract upgrade with adjusted thresholds if gas schedule changes significantly.
+
+### FM2: Storage Slot Collision
+
+- **Description:** The checksum calculation for message validation uses storage slots that could potentially collide with other slots from the access list, leading to false validation of messages.
+- **Risk Assessment:** High
+  - Potential Impact: Critical. Storage collisions could allow unauthorized messages to be validated.
```

Comment:
The impact of this is that we would not be able to filter messages at ingress that created invalid executing messages. This creates a spam incentive for MEV searchers that can attempt an arb and create an invalid executing message at no cost (guarantee of no inclusion)